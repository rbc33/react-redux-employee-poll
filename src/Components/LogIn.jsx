import { useState } from "react";
import { setAuthedUser } from "../actions/authedUser";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";

const LogIn = ({ users, dispatch }) => {
	const [selectedUser, setSelectedUser] = useState("");
	const navigate = useNavigate();

	const handleSubmit = (e) => {
		e.preventDefault();
		if (selectedUser) {
			dispatch(setAuthedUser(selectedUser));
			navigate("/");
		}
	};

	return (
		<div className="center">
			<h1>Employee Polls</h1>
			<img
				className="logIn-image"
				src="https://i.blogs.es/f32438/1366_2000-5-/1024_2000.jpeg"
			/>
			<h1>Log In</h1>
			<form onSubmit={handleSubmit}>
				<select
					className="selector"
					value={selectedUser}
					onChange={(e) => setSelectedUser(e.target.value)}
				>
					<option value="">Select a user</option>
					{Object.values(users).map((user) => (
						<option key={user.id} value={user.id}>
							{user.name}
						</option>
					))}
				</select>
				<br />
				<button className="btn" type="submit" disabled={!selectedUser}>
					Log In
				</button>
			</form>
		</div>
	);
};
const mapStateToProps = (state) => ({
	users: state.users,
	dispatch: state.dispatch,
});

export default connect(mapStateToProps)(LogIn);
