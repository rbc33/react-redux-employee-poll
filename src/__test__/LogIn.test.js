import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import LogIn from "../Components/LogIn";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import { thunk } from "redux-thunk";
import { expect, it, describe, beforeEach } from "@jest/globals";

const mockStore = configureStore([thunk]);

describe("Login Component", () => {
	let store;

	const initialState = {
		users: {
			testuser: {
				id: "testuser",
				name: "Test User",
				answers: {},
				questions: [],
			},
		},
		questions: {},
		loadingBar: { default: 0 },
	};

	beforeEach(() => {
		store = mockStore({
			...initialState,
			authedUser: null,
		});
	});

	it("renders login component correctly", () => {
		render(
			<Provider store={store}>
				<BrowserRouter>
					<LogIn />
				</BrowserRouter>
			</Provider>
		);

		expect(screen.getByTestId("login-component")).toBeInTheDocument();
	});

	it("allows user selection", () => {
		render(
			<Provider store={store}>
				<BrowserRouter>
					<LogIn />
				</BrowserRouter>
			</Provider>
		);

		const selectElement = screen.getByRole("combobox");
		fireEvent.change(selectElement, { target: { value: "testuser" } });

		expect(selectElement.value).toBe("testuser");
		expect(selectElement.value).toM;
	});
	it("will match snapshot when no name is selected", () => {
		const component = render(
			<Provider store={store}>
				<BrowserRouter>
					<LogIn />
				</BrowserRouter>
			</Provider>
		);
		expect(component).toMatchSnapshot();
	});
});
