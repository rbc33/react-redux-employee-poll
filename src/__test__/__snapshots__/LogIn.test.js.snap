// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component will match snapshot when no name is selected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="center"
        data-testid="login-component"
      >
        <h1>
          Employee Polls
        </h1>
        <img
          class="logIn-image"
          src="https://i.blogs.es/f32438/1366_2000-5-/1024_2000.jpeg"
        />
        <h1>
          Log In
        </h1>
        <form>
          <select
            class="selector"
          >
            <option
              value=""
            >
              Select a user
            </option>
            <option
              value="testuser"
            >
              Test User
            </option>
          </select>
          <br />
          <button
            class="btn"
            disabled=""
            type="submit"
          >
            Log In
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="center"
      data-testid="login-component"
    >
      <h1>
        Employee Polls
      </h1>
      <img
        class="logIn-image"
        src="https://i.blogs.es/f32438/1366_2000-5-/1024_2000.jpeg"
      />
      <h1>
        Log In
      </h1>
      <form>
        <select
          class="selector"
        >
          <option
            value=""
          >
            Select a user
          </option>
          <option
            value="testuser"
          >
            Test User
          </option>
        </select>
        <br />
        <button
          class="btn"
          disabled=""
          type="submit"
        >
          Log In
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
